test/jdk/java/awt/Dialog/JaWSTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2002, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4690465
+  @summary Tests that after dialog is hidden on another EDT, owning EDT gets notified.
+  @modules java.desktop/sun.awt
+  @key headful
+  @run main JaWSTest
+*/
+
+import java.awt.Button;
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import sun.awt.SunToolkit;
+import sun.awt.AppContext;
+
+public class JaWSTest implements ActionListener, Runnable {
+
+    static volatile Frame frame;
+    static volatile JaWSTest worker;
+    static volatile Dialog dummyDialog;
+    static final Object signalObject = new Object();
+    static volatile AppContext appContextObject = null;
+    static volatile Button button = null;
+    static final CountDownLatch dialogFinished = new CountDownLatch(1);
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(JaWSTest::createUI);
+            Robot robot = new Robot();
+            robot.waitForIdle();
+            robot.delay(1000);
+            Point buttonLocation = button.getLocationOnScreen();
+            robot.mouseMove(buttonLocation.x + button.getWidth()/2,
+                            buttonLocation.y + button.getHeight()/2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            if (!worker.dialogFinished.await(5, TimeUnit.SECONDS)) {
```

Comment:
```suggestion
            if (!dialogFinished.await(5, TimeUnit.SECONDS)) {
```
Since `dialogFinished` is static, it can be accessed directly, without the `worker` object.